src/index.d.ts

```diff
@@ -31,10 +47,6 @@ declare module 'material-ui-search-bar' {
        * Override the search icon.
        */
       searchIcon?: any;
```

Comment:
Is there a type for "react element"? :thinking:

Reply:
We could use "JSX.Element"

Reply:
Material-UI uses `React.ReactNode` for children and similar props. Seems to be a good option here.

Reply:
That allows values such as searchIcon="hello world" though

Reply:
Allright, it should be `Icon | SvgIcon` from Material-UI, both need to be imported.

Reply:
Or maybe `React.ReactType<Icon> | React.ReactType<SvgIcon>`? I've never used TypeScript for React development.

Reply:
Does that means the prop won't allow values such as <button>Search</button>?

Reply:
Indeed. The `icon` prop expects an actual icon, not a string. :wink:

Reply:
Allright, let's settle with `React.ReactNode` then. If people are using strings as an icon, that's not my fault. :roll_eyes:

Reply:
I'm not in favor of using React.ReactNode. I'm proposing JSX.Element as the solution

`React.ReactNode` would allow `searchIcon="hello world"` which actually results in an error.

`JSX.Element` would allow any elements including Material UI icons, custom elements, and elements from other libraries.

Reply:
@gvso Fine.